src/core_plugins/metrics/server/lib/vis_data/request_processors/table/date_histogram.js

```diff
@@ -0,0 +1,26 @@
+import _ from 'lodash';
+import getBucketSize from '../../helpers/get_bucket_size';
+import getIntervalAndTimefield from '../../get_interval_and_timefield';
+import getTimerange from '../../helpers/get_timerange';
+import { calculateAggRoot } from './calculate_agg_root';
+
+export default function dateHistogram(req, panel) {
+  return next => doc => {
+    const { timeField, interval } = getIntervalAndTimefield(panel);
+    const { bucketSize, intervalString } = getBucketSize(req, interval);
+    const { from, to }  = getTimerange(req);
+    panel.series.forEach(column => {
+      const aggRoot = calculateAggRoot(doc, column);
+      _.set(doc, `${aggRoot}.timeseries.date_histogram`, {
+        field: timeField,
+        interval: intervalString,
+        min_doc_count: 0,
+        extended_bounds: {
+          min: from.valueOf(),
+          max: to.valueOf() - (bucketSize * 1000)
```

Comment:
@simianhacker why do you reduce the `bucketSize` here?

Reply:
This was removed from the rest of TSVB but because this PR languished in review for so long I forgot to remove it from here as well. I have a PR that fixes this. #14553

Reply:
Thnx, should probably be removed from here as well no? https://github.com/elastic/kibana/pull/12813/files/abeeaa7224fd391aa73e3b1200888da3623e725c#diff-63d4f38a0a2ce1a25e2e18de8982c1b1R21

Reply:
Oops... Added that one as well to the PR... Thanks!